packages/patternfly-4/react-charts/src/components/ChartDonutUtilization/ChartDonutUtilization.tsx

```diff
@@ -541,14 +549,20 @@ export const ChartDonutUtilization: React.FunctionComponent<ChartDonutUtilizatio
     if (data) {
       const datum = getData();
       const donutThresholds = getDonutThresholds();
+      const mergeThemeProps = (i: number) => {
+        // Merge just the first color of dynamic (blue, green, etc.) with static (gray) for expected colorScale
+        newTheme.pie.colorScale[0] = donutThresholds[i].color;
+        newTheme.legend.colorScale[0] = donutThresholds[i].color;
+      };
       for (let i = 0; i < donutThresholds.length; i++) {
```

Comment:
I think we'll have to enforce sorting on the thresholds array. They need to be sorted in ascending order by the value prop when `inverted === false`, and descending when `inverted === true`.

It's possible to pass a `thresholds` prop in an unsorted or reverse sorted order, which would break this function.

For example:
```js
[
  {
    value: 90, 
    color: 'red'
  }, 
  {
    value: 60, 
    color: 'yellow'
  }
]
```

If a value is greater than 90, we would never get the 'red' color because we would iterate past the `value: 90` threshold, see that the data value is greater than 60 as well, and colorScale[0] would be set to yellow.

Reply:
I'll open a separate issue for this and it can be fixed as a follow on since it was an existing issue before this PR.